psutil/tests/__init__.py

```diff
@@ -131,6 +131,8 @@
 COVERAGE = 'COVERAGE_RUN' in os.environ
 # are we a 64 bit process?
 IS_64BIT = sys.maxsize > 2 ** 32
+# PEP 405 virtual environment
+IS_VIRTUALENV = sys.prefix != getattr(sys, "base_prefix", sys.prefix)
```

Comment:
I'm afraid this code won't work on Python 2.7 (`base_prefix` was added in [3.3](https://docs.python.org/3/library/sys.html#sys.base_prefix)).

A similar check is present in the Makefile, but we do something different (I'm not sure this works on 3.X though):
https://github.com/giampaolo/psutil/blob/2974941c179c1637b5102b0cebd3d332c2847678/Makefile#L40-L41

I'm surprised detecting a venv is so difficult. FWIW, I bumped into this:
https://stackoverflow.com/a/1883251
...which seems to work on both Python 2 and 3.

Reply:
We only need to detect this properly on python 3.7+ on windows.
If python 2.7 does not respect PEP 405, we won't detect it's running in a virtual environment but the test outcome will be correct nonetheless.

Reply:
I removed the variable IS_VIRTUALENV to lessen confusion that this could trigger.